### PR TITLE
[READY] Constantinople tests 

### PR DESCRIPTION
**Pull request by winsvega**
implement new test cases and point cpp to latest test branch

- [x] stCreate2
- [x] stExtcodehash
- [ ] stExtcodeSize (cloned from stExtcodehash but for extcodesize) (could be done in the same test files)
- [ ] stExtcodeCopy (cloned from stExtcodeHash but for extcodesize) (could be done in the same test files)
- [ ] bcTransitionTests to Constantinople
  - [x] simple transition test. check difficulty. check difficulty change with uncle and reward change.
  - [ ] more tests? 
- [x] refill all tests
- [x] more Sstore tests
- [ ] bit shift test check
- [x] BasicTests/difficultyConstantinople
- [ ] make tests for bugreports from libfuzzer and fuzzed test generator

### CODE REVIEW

**Comment by gumb0** [comment]
What about `BOOST_AUTO_TEST_CASE(stCreate2) {}` in `BlockChainTests.cpp` ?

**Comment by winsvega** [comment]
Yes. There will be stCreate2, extcodehash, and blockchain transition tests

File: test/tools/jsontests/BlockChainTests.cpp [hunk 1]
```diff
@@ -654,7 +654,7 @@ void overwriteBlockHeaderForTest(mObject const& _blObj, TestBlock& _block, Chain
             ho.count("number") ? toU256(ho["number"]) : header.number(),
             ho.count("gasLimit") ? toU256(ho["gasLimit"]) : header.gasLimit(),
             ho.count("gasUsed") ? toU256(ho["gasUsed"]) : header.gasUsed(),
-            ho.count("timestamp") ? toU256(ho["timestamp"]) : header.timestamp(),
+            ho.count("timestamp") ? toU256(ho["timestamp"]) : header.timestamp() > 0 ? header.timestamp() : 1000,
```
Comment (gumb0) [inline test/tools/jsontests/BlockChainTests.cpp]: What's wrong with timestamp 0?

File: libethcore/SealEngine.cpp [hunk 2]
```diff
@@ -66,7 +66,7 @@ void SealEngineFace::verify(Strictness _s, BlockHeader const& _bi, BlockHeader c
 {
     _bi.verify(_s, _parent, _block);
 
-    if (_s != CheckNothingNew)
+    if (_s != CheckNothingNew && _s != JustSeal)
```
Comment (gumb0) [inline libethcore/SealEngine.cpp]: It doesn't seem to me that `JustSeal` works right (e.g. `Ethash` validates seal only if `_s == CheckEverything`)
It looks to be used only in json tests, do you actually need it? Couldn't you use `CheckNothingNew` ?

Comment (winsvega) [reply libethcore/SealEngine.cpp]: I could. But logically those checks should not be performed for JustSeal and for lots of other enums as well

Comment (gumb0) [reply libethcore/SealEngine.cpp]: I propose to remove `JustSeal` from enum then.

**Comment by gumb0** [comment]
Can we merge this anytime soon? What's blocking it?

**Comment by winsvega** [comment]
We could merge it after rebase on top of remove strictness:justseal

**Comment by winsvega** [comment]
(ready to merge)

**Pull request merged**